components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/context/model/Flow.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core.context.model;
+
+import java.util.EnumSet;
+
+/**
+ * Enum for flows.
+ * This enum holds the flow type and the action that is being executed.
+ */
+public class Flow {
+
+    /**
+     * Enum for types.
+     */
+    public enum Type {
+
+        PASSWORD_RECOVERY(EnumSet.of(Action.INVITE, Action.UPDATE, Action.RESET));
```

Comment:
This we will have to rethink. Having flow types and actions overall complicate the understanding of the exact flow

Reply:
Fixed